chat.js

```diff
@@ -920,25 +925,38 @@ Chat.uncacheTree = function (root) {
 	} while (uncache.length > 0);
 };
 
-Chat.loadCommands = function () {
-	if (Chat.commands) return;
+Chat.loadPlugins = function () {
+	if (Chat.commands || Chat.filter) return;
 
 	FS('package.json').readTextIfExists().then(data => {
 		if (data) Chat.package = JSON.parse(data);
 	});
 
 	let baseCommands = Chat.baseCommands = require('./chat-commands').commands;
 	let commands = Chat.commands = Object.assign({}, baseCommands);
+	let chatfilters = [];
 
-	// Install plug-in commands
+	// Install plug-in commands and chat filter
 
 	// info always goes first so other plugins can shadow it
 	Object.assign(commands, require('./chat-plugins/info').commands);
 
 	for (let file of FS('chat-plugins/').readdirSync()) {
 		if (file.substr(-3) !== '.js' || file === 'info.js') continue;
 		Object.assign(commands, require('./chat-plugins/' + file).commands);
+
+		const filter = require(`./chat-plugins/${file}`).chatfilter;
+		if (filter) chatfilters.push(filter);
 	}
+
+	Chat.filter = function (message, user, room, connection, targetUser) {
+		if (!chatfilters.length) return message;
+
+		for (let i = 0; i < chatfilters.length; i++) {
+			const output = chatfilters[i].call(this, message, user, room, connection, targetUser);
+			if (typeof output === 'string') return message;
```

Comment:
Because this is the implementation of a core feature intended to accomodate plugins.

Restricting the core so that plugins will only be able to do what we already would be shortsighted and pointless.